### PR TITLE
Update release-index.json schema

### PR DESCRIPTION
**Pull request by richlander**
How does this look? Is this a good direction?

In reference to: https://github.com/dotnet/core/pull/9427#discussion_r1689298435

@martincostello @baronfel @Falco20019

### CODE REVIEW

File: release-notes/releases-index.json [hunk 1]
```diff
@@ -13,7 +13,11 @@
             "eol-date": "2026-05-11",
             "release-type": "sts",
             "releases.json": "https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/9.0/releases.json",
-            "supported-os.json": "https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/9.0/supported-os.json"
+            "patch-releases-info-uri": "https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/9.0/releases.json",
+            "patch-releases-index-uri": "https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/9.0/patch-releases-index.json",
+            "supported-os.json": "https://dotnetcli.blob.core.windows.net/dotnet/release-metadata/9.0/supported-os.json",
```
Comment (martincostello) [inline release-notes/releases-index.json:18]: Dupe property.

```suggestion
```

**Review by martincostello — APPROVED** [review]
LGTM

File: release-notes/releases-index.json (hunk 1 above)
Comment (richlander) [reply release-notes/releases-index.json:18]: Are you saying it is a dupe with line 19? Line 15 and line 18 have to be kept for compat.

Comment (martincostello) [reply release-notes/releases-index.json:18]: Sorry - this was me being an idiot and reading the diff incorrectly 🤦

Comment (richlander) [reply release-notes/releases-index.json:18]: All good.

**Comment by richlander** [comment]
I'm going to close this PR. It was intended to drive a design conversation. I'll wait until we can generate this file.

If these is more feedback, we can definitely continue the conversation. I just want to make it clear that I don't intend to merge these changes via this PR.

**Pull request closed without merging**